client/src/main/java/io/confluent/kafka/schemaregistry/client/MockSchemaRegistryClient.java

```diff
@@ -517,9 +522,9 @@ public synchronized List<Integer> deleteSubject(
       throws IOException, RestClientException {
     schemaCache.remove(subject);
     idCache.remove(subject);
-    versionCache.remove(subject);
+    Map<ParsedSchema, Integer> versions = versionCache.remove(subject);
     compatibilityCache.remove(subject);
-    return Collections.singletonList(0);
+    return versions.values().stream().sorted().collect(Collectors.toList());
```

Comment:
@rayokota this is breaking ksqlDB's use of the MockSchemaRegistry.  ksqlDB calls for a soft delete and then a hard delete in some cases.  Since the soft delete removes the subject from the versionCache, the second call throws a NPE.